stack/stack.go

```diff
@@ -0,0 +1,252 @@
+	"flutter":      "osx-vs4mac-stable",
+	"android":      "linux-docker-android",
+	"macos":        "osx-xcode-10.0.x",
+	"ios":          "osx-xcode-10.0.x",
+}
+
+// optionsStacks is the array of all available stacks.
+//
+// Deprecated: please use the Stacks member defined in this class.
+var optionsStacks = []string{
+	"linux-docker-android-lts",
+	"linux-docker-android",
+	"osx-vs4mac-beta",
+	"osx-vs4mac-previous-stable",
+	"osx-vs4mac-stable",
+	"osx-xamarin-stable",
+	"osx-xcode-11.0.x",
+	"osx-xcode-11.1.x",
+	"osx-xcode-11.2.x",
+	"osx-xcode-10.0.x",
+	"osx-xcode-10.1.x",
+	"osx-xcode-10.2.x",
+	"osx-xcode-8.3.x",
+	"osx-xcode-9.4.x",
+	"osx-xcode-edge",
+}
+
+// StackOptionsMap holds the available stack for the given platforms, where the first item is the default.
+//
+// This should be in sync with: https://github.com/bitrise-io/bitrise-website/blob/master/config/available_stacks.yml
+var StackOptionsMap = map[Platform][]Stack{
+	Xamarin: {
+		OsxVs4MACStable,
+		OsxVs4MACBeta,
+		OsxVs4MACPreviousStable},
+	Cordova: {
+		OsxXcode102X,
+		LinuxDockerAndroidLts,
+		LinuxDockerAndroid,
+		OsxVs4MACStable,
+		OsxVs4MACBeta,
+		OsxVs4MACPreviousStable,
+		OsxXcode112X,
+		OsxXcode111X,
+		OsxXcode110X,
+		OsxXcode101X,
+		OsxXcode100X,
+		OsxXcode94X,
+		OsxXcode83X,
+		OsxXcodeEdge},
+	ReactNative: {
+		OsxXcode102X,
+		LinuxDockerAndroidLts,
+		LinuxDockerAndroid,
+		OsxVs4MACStable,
+		OsxVs4MACBeta,
+		OsxVs4MACPreviousStable,
+		OsxXcode112X,
+		OsxXcode111X,
+		OsxXcode110X,
+		OsxXcode101X,
+		OsxXcode100X,
+		OsxXcode94X,
+		OsxXcode83X,
+		OsxXcodeEdge},
+	Ionic: {
+		OsxXcode102X,
+		LinuxDockerAndroidLts,
+		LinuxDockerAndroid,
+		OsxVs4MACStable,
+		OsxVs4MACBeta,
+		OsxVs4MACPreviousStable,
+		OsxXcode112X,
+		OsxXcode111X,
+		OsxXcode110X,
+		OsxXcode101X,
+		OsxXcode100X,
+		OsxXcode94X,
+		OsxXcode83X,
+		OsxXcodeEdge,
+	},
+	Flutter: {
+		OsxXcode102X,
+		LinuxDockerAndroidLts,
+		LinuxDockerAndroid,
+		OsxVs4MACStable,
+		OsxVs4MACBeta,
+		OsxVs4MACPreviousStable,
+		OsxXcode112X,
+		OsxXcode111X,
+		OsxXcode110X,
+		OsxXcode101X,
+		OsxXcode100X,
+		OsxXcode94X,
+		OsxXcode83X,
+		OsxXcodeEdge,
+	},
+	Android: {
+		LinuxDockerAndroid,
+		LinuxDockerAndroidLts,
+		OsxVs4MACStable,
+		OsxVs4MACBeta,
+		OsxVs4MACPreviousStable,
+		OsxXcode112X,
+		OsxXcode111X,
+		OsxXcode110X,
+		OsxXcode102X,
+		OsxXcode101X,
+		OsxXcode100X,
+		OsxXcode94X,
+		OsxXcode83X,
+		OsxXcodeEdge,
+	},
+	MacOS: {
+		OsxXcode102X,
+		OsxVs4MACStable,
+		OsxVs4MACBeta,
+		OsxVs4MACPreviousStable,
+		OsxXcode112X,
+		OsxXcode111X,
+		OsxXcode110X,
+		OsxXcode101X,
+		OsxXcode100X,
+		OsxXcode94X,
+		OsxXcode83X,
+		OsxXcodeEdge},
+	Ios: {
+		OsxXcode102X,
+		OsxVs4MACStable,
+		OsxVs4MACBeta,
+		OsxVs4MACPreviousStable,
+		OsxXcode112X,
+		OsxXcode111X,
+		OsxXcode110X,
+		OsxXcode101X,
+		OsxXcode100X,
+		OsxXcode94X,
+		OsxXcode83X,
+		OsxXcodeEdge},
+}
+
+// Stack defines a Stack with it's name.
+type Stack string
+
+const (
```

Comment:
I think in a structure like os-tool-version might be better.
osx.xcode.112x
osx.vs4mac.stable

etc.

Will work like now but for me it is a bit of too much hardcoding for a relatively dynamic stuff.

Reply:
Updated to what we discussed.